### ISSUE TITLE
App crash on ios 15.4

### ISSUE THREAD

**Issue by anija** [react-native-maps/react-native-maps#4130]
# Bug report

## Summary
App started to crash with ios 15.4

## Environment info

react-native: 0.66.4
react: 17.0.2
react-native-maps: 0.30.1

Target Platform: iOS - 15.4 Simulator/iPhone 13
Using GoogleMaps

`react-native info` output:

System:
    OS: macOS 12.2.1
    CPU: (8) arm64 Apple M1
    Memory: 1.02 GB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.2 - ~/.nvm/versions/node/v16.13.2/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.13.2/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.2/bin/npm
    Watchman: 2022.01.24.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.2 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.4, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7678000
    Xcode: 13.3/13E113 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.12 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.2 => 17.0.2 
    react-native: 0.66.4 => 0.66.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

## Steps to reproduce

1. Create a component with:
```import React, { useEffect, useState } from 'react';
import { StyleSheet } from 'react-native';
import MapView, { PROVIDER_GOOGLE } from 'react-native-maps';

<MapView
  provider={PROVIDER_GOOGLE}
  style={StyleSheet.absoluteFill}
/>
```
2. Display the component
3. After a few seconds, the app crashes

Describe what you expected to happen:

1. The app don't crash

Everything works fine when provider is not set.

Crash report: [crash_report.txt](https://github.com/react-native-maps/react-native-maps/files/8268928/crash_report.txt)

**Comment by Simon-TechForm** [comment]
I'm unable to reproduce this. Is this happening in a clean react native project?

**Comment by Simon-TechForm** [comment]
So it seems like this might be M1 specific, which could be the reason why I couldn't reproduce it. We should keep an eye on this: https://issuetracker.google.com/issues/224584852

**Comment by AlMel3000** [comment]
App crashes with provider={PROVIDER_GOOGLE} on M1

When I comment remove MapView from render - app works as expected

Sentry shows me that:

EXC_BAD_ACCESS
Exception 1, Code 2768238912, Subcode 8 >
Attempted to dereference garbage pointer 0x7ff4a4fff940.


<img width="600" alt="Screenshot 2022-03-21 at 16 07 37" src="https://user-images.githubusercontent.com/19572431/159302526-5e70ef9b-c93f-41e1-b203-6cad62b07d2d.png">

**Comment by JackJBlundell** [comment]
I too am experiencing this issue but on iOS 15.0 - Sometimes the Map will render for me, but mostly it just crashes with the only warning output being: 

```
Mapbox warning Falling back to MGLIdeographsRasterizedLocally because  MGLIdeographicFontFamilies are specified. {"filePath": "-[MGLRendererConfiguration glyphsRasterizationOptionsWithLocalFontFamilyName:rasterizationMode:]", "level": "warning", "line": 111, "message": "Falling back to MGLIdeographsRasterizedLocally because  MGLIdeographicFontFamilies are specified."}
```
From online i've seen this isn't the issue though.

Dependencies:
```
"dependencies": {
    "@homee/react-native-mapbox-navigation": "^1.1.0",
    "@react-native-community/datetimepicker": "^6.0.1",
    "@react-native-community/masked-view": "^0.1.11",
    "@react-native-mapbox-gl/maps": "^8.6.0-beta.0",
    "@react-native-picker/picker": "^2.3.1",
    "@react-navigation/native": "^6.0.8",
    "@react-navigation/stack": "^6.1.1",
    "@types/firebase": "^3.2.1",
    "@types/mapbox": "^1.6.42",
    "@types/react-geocode": "^0.2.0",
    "@types/react-native-vector-icons": "^6.4.10",
    "@types/turf": "^3.5.32",
    "firebase": "^8.3.0",
    "mapbox": "^1.0.0-beta10",
    "moment": "^2.29.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-geocode": "^0.2.3",
    "react-native": "^0.63.4",
    "react-native-elements": "^3.4.2",
    "react-native-geolocation-service": "^5.3.0-beta.4",
    "react-native-gesture-handler": "^2.3.1",
    "react-native-google-places-autocomplete": "^2.4.1",
    "react-native-image-picker": "^4.7.3",
    "react-native-permissions": "^3.3.1",
    "react-native-phone-call": "^1.0.9",
    "react-native-reanimated": "^2.4.1",
    "react-native-safe-area-context": "^3.1.9",
    "react-native-screens": "^3.13.1",
    "react-native-vector-icons": "^9.1.0",
    "rn-sliding-up-panel": "^2.4.5",
    "turf": "^3.0.14",
    "url": "^0.11.0"
```
Podfile:
```
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '11.0'
install! 'cocoapods', :disable_input_output_paths => true

target 'RouteBuddies' do
  config = use_native_modules!
  
  use_frameworks!
  use_react_native!(
    :path => config[:reactNativePath],
    # to enable hermes on iOS, change `false` to `true` and then install pods
    :hermes_enabled => false
  )

  permissions_path = '../node_modules/react-native-permissions/ios'

  pod 'Permission-Camera', :path => "#{permissions_path}/Camera"
  pod 'Permission-LocationAccuracy', :path => "#{permissions_path}/LocationAccuracy"
  pod 'Permission-LocationAlways', :path => "#{permissions_path}/LocationAlways"
  pod 'Permission-LocationWhenInUse', :path => "#{permissions_path}/LocationWhenInUse"
  pod 'Permission-Notifications', :path => "#{permissions_path}/Notifications"


  target 'RouteBuddiesTests' do
    inherit! :complete
    # Pods for testing
    
    $RNMapboxMapsImpl = 'mapbox'
    $RNMapboxMapsVersion = '~> 6.3'
    $ReactNativeMapboxGLIOSVersion = "~> 6.3"
    ENV['REACT_NATIVE_MAPBOX_GL_USE_FRAMEWORKS'] = 'true'
    
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable the next line.
  # use_flipper!({ 'Flipper-Folly' => '2.3.0' })

  if (defined?(TargetsToChangeToDynamic)).nil?
  TargetsToChangeToDynamic = ['MapboxMobileEvents']
  elsif TargetsToChangeToDynamic.include? 'MapboxMobileEvents' === false
    TargetsToChangeToDynamic.push('MapboxMobileEvents')
  end
  $dynamic_framework = ['MapboxMobileEvents']
  pod 'react-native-mapbox-gl', :path => '../node_modules/@react-native-mapbox-gl/maps'
  pod 'MapboxNavigation', '~> 1.3'

  pre_install do |installer|
    $RNMapboxMaps.pre_install(installer)
  end
  pre_install do |installer|
    $RNMBNAV.pre_install(installer)
  end
  pre_install do |installer|

    $RNMBGL.pre_install(installer)
    Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
    installer.pod_targets.each do |pod|
      if !$dynamic_framework.include?(pod.name)
        def pod.build_type;
          Pod::BuildType.static_library
        end
      end
    end
  end

  post_install do |installer|
    # This below is causing issue

    $RNMBNAV.post_install(installer)
    installer.pods_project.targets.each do |target|
      target.build_configurations.each do |config|
      # some older pods don't support some architectures, anything over iOS 11 resolves that
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '12.0'
      end
    end
  end
  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'
end
```

**Comment by arnekolja** [comment]
Having this crash on all iOS 15 versions in Simulator (not sure if on real device too) since I changed npm dependencies lately. Map is shown for a moment, but then gets distorted textures from the left and crashes the app.
Removing PROVIDER_GOOGLE helped for me, too.

**Comment by leighquince** [comment]
Seeing the same behviour on simulator;
Non M1 mac
ios version = 15.4
"react-native-maps": "0.30.1",
"react-native": "0.65.1",

Crashes only in simulator when `provider={PROVIDER_GOOGLE}` is set

**Comment by salah-ghanim** [comment]
I see the following ticket on google map's iOS SDK https://github.com/googlemaps/google-maps-ios-utils/issues/398
and seems like a crash is reported for iOS 15.4

since this is a dependency issue that we can't do anything about I would recommend ether:
1. use another OS version of the simulator 
2. force an older version of the SDK that doesn't include the latest GoogleMaps SDK

From time to time such issues with Google maps appear and usually they're fixed in the next release.

At the same time we regularly update dependencies for this project so the fix will eventually land on the latest release

Because of this I would like to close this issue and keep only issues that can be worked on open, so that the maintainers can focus on them.

**Comment by salah-ghanim** [comment]
@Simon-TechForm thanks for your input. yes I don't mind keeping this open if we add a special tag here (waiting on dependency update) for example , just to reduce the cognitive load involved with maintaining this lib.

let's reopen this anyway and discuss a solution that we can use as maintainers moving forward.

**Comment by erickbennett** [comment]
@Simon-TechForm 

I completely agree. I am also having the issue.

**Comment by loydchristmass** [comment]
Having this issue also on Latest 0.30.1, and simulator running 15.4.

**Comment by ajyku** [comment]
I have also faced the same issue in the iOS 15.4 simulator.

**Comment by kigh143** [comment]
Hey guys, I am getting an error for maximum-call-stack-size-exceeded, when I comment out the mapView, it works fine. At first, it was just crashing.

**Comment by shakawy** [comment]
Also facing the same issue in the iOS 15.4 simulator.
removing `provider={PROVIDER_GOOGLE}` solves the issue.
but i don't like this temporary solution cause i want to use google maps view

**Comment by gmztech** [comment]
I was having this issue and i manage to solve it this way. 
`import MapView, {PROVIDER_GOOGLE, PROVIDER_DEFAULT} from 'react-native-maps';`
Then create an useState for the map provider and set its initial value to the default provider (PROVIDER_DEFAULT), then on the MapView component, use the onMapReady listener, inside of it change the state of the provider to the PROVIDER_GOOGLE and it should load without crashing after ✌️

**Comment by AlMel3000** [comment]
@gmztech 
> I was having this issue and i manage to solve it this way.
> `import MapView, {PROVIDER_GOOGLE, PROVIDER_DEFAULT} from 'react-native-maps';`
> Then create an useState for the map provider and set its initial value to the default provider (PROVIDER_DEFAULT), then on the MapView component, use the onMapReady listener, inside of it change the state of the provider to the PROVIDER_GOOGLE and it should load without crashing after ✌️

Great solution, but after 0.28.0 version onMapReady not calling on iOS 😂
 (here is a separate issue for this)

**Comment by shakawy** [comment]
Ok i managed to use `PROVIDER_GOOGLE`
through setting the height of `<MapView />` style.
I was calculating the height dynamically and it was `100.28299999999999` so i think this is exceeding the data type size, when i rounded the number everything worked find without any crashes

**Comment by leeeGreat** [comment]
> 
这个大神太赞了，不崩溃了！

**Comment by leeeGreat** [comment]
> @gmztech
> 
> > I was having this issue and i manage to solve it this way.
> > `import MapView, {PROVIDER_GOOGLE, PROVIDER_DEFAULT} from 'react-native-maps';`
> > Then create an useState for the map provider and set its initial value to the default provider (PROVIDER_DEFAULT), then on the MapView component, use the onMapReady listener, inside of it change the state of the provider to the PROVIDER_GOOGLE and it should load without crashing after ✌️
> 
> Great solution, but after 0.28.0 version onMapReady not calling on iOS 😂 (here is a separate issue for this)

version 0.31.1, iOS run ok，that works for me

**Comment by umarabdulhuq** [comment]
I have same issue solved it by giving static width and height like (width:500,height:500) or by giving height and width with Dimensions from react-native, use flex:1 also leads to app crash on iOS.

**Comment by Hayk-Zohrabyan** [comment]
@umarabdulhuq Thank you so much.
It really works. ))

**Comment by jbrodriguez** [comment]
> @ravirajn22 would you be able to supply a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? I'm struggling to reproduce this. https://issuetracker.google.com/issues/224584852 still seems highly relevant.

@salah-ghanim in that issue and [this one](https://issuetracker.google.com/issues/221643638), they mention they won't fix it and suggest to switch to metal renderer.

in the first thread, a user reported that doing that made the map stable again, now, how can one switch to the metal renderer ?

**Comment by jbrodriguez** [comment]
hi, i opened [the PR above](https://github.com/react-native-maps/react-native-maps/pull/4334) hoping someone can help figure out the run on main thread error, but hopefully this fixes the issue by enabling metal renderer

**Comment by Eli-Nathan** [comment]
I'm still seeing this issue.

React native version: 0.69.4
React native maps version: 1.2.0

The onMapReady solution does not work for me and neither does the width/height solution.
Constant crashes unless I change the provider. Any update on this?

**Comment by yutaro47** [comment]
@Eli-Nathan me too

**Comment by FabrizioCaldarelli** [comment]
I solved setting a style to marker containing width or height as you prefer, such as:

```
  <Marker
      key={marker.id}
      coordinate={marker.coordinate}
      title={marker.title}
      description={marker.description}
      style={{width: 250}}
    image={Images.pinMap}>
```

**Comment by isaacparsons-dw** [comment]
@Eli-Nathan @yutaro47 I have the same issue too - both solutions don't work. Have either of you managed to rectify?

**Comment by december1990** [comment]
> I have same issue solved it by giving static width and height like (width:500,height:500) or by giving height and width with Dimensions from react-native, use flex:1 also leads to app crash on iOS.

for me just decreases the number of times this issue happens

**Comment by AdeelCyber** [comment]
I had the same issue. I found a solution online.. Just add 
android:{
......,
"config": {
        "googleMaps": {
          "apiKey": <MAP API KEY>
        }
      }
}
and for IOS
ios:{
.....,
"config": {
        "googleMapsApiKey": <MAP API KEY>
      }
}

**Comment by ornnikarn** [comment]
> > I have same issue solved it by giving static width and height like (width:500,height:500) or by giving height and width with Dimensions from react-native, use flex:1 also leads to app crash on iOS.
> 
> for me just decreases the number of times this issue happens

same for me

**Comment by MRadek55** [comment]
Guys, did u add these lines? 

```
# React Native Maps dependencies
# The following line is only needed if building on an Apple silicon Mac without rosetta.
pod 'Google-Maps-iOS-Utils', :git => 'https://github.com/Simon-TechForm/google-maps-ios-utils.git', :branch => 'feat/support-apple-silicon'

rn_maps_path = '../node_modules/react-native-maps'
pod 'react-native-google-maps', :path => rn_maps_path
```

I didn't put it and I had same problem as u.

**Comment by SzymonEasyOZE** [comment]
Still the issue on iOS 16.1
